core/services/job/orm.go

```diff
@@ -846,42 +846,100 @@ func (o *orm) PipelineRunsByJobsIDs(ids []int32) (runs []pipeline.Run, err error
 	return runs, errors.Wrap(err, "PipelineRunsByJobsIDs failed")
 }
 
+func (o *orm) loadPipelineRunIDs(jobID *int32, offset, limit int, tx pg.Queryer) (ids []int64, err error) {
+	var filter string
+	lggr := logger.Sugared(o.lggr)
+
+	if jobID != nil {
+		filter = fmt.Sprintf("JOIN jobs USING(pipeline_spec_id) WHERE jobs.id = %d AND ", *jobID)
+	} else {
+		filter = "WHERE "
+	}
+	stmt := fmt.Sprintf(`SELECT p.id FROM pipeline_runs AS p %s p.id >= $3 AND p.id <= $4
+			ORDER BY p.id DESC OFFSET $1 LIMIT $2`, filter)
+	var res sql.NullInt64
+
+	if err = tx.Get(&res, "SELECT MAX(id) FROM pipeline_runs"); err != nil {
+		err = errors.Wrap(err, "error while loading runs")
+		return
+	} else if !res.Valid {
+		// MAX() will return NULL if there are no rows in table.  This is not an error
+		return
+	}
+
+	maxID := res.Int64
```

Comment:
Would it make sense to do this block first, since the `stmt` constructed from the `filter` is not used until later on?